WBooks/BookDetailViewController.swift

```diff
@@ -5,30 +5,35 @@
 //  Created by Matías David Schwalb on 30/05/2019.
 //  Copyright © 2019 Wolox. All rights reserved.
 //
-
+import ReactiveSwift
+import Result
+import ReactiveCocoa
 import UIKit
 
 final class BookDetailViewController: UIViewController {
-
-    private let bookID: Int
+    
     private let bookDetailView: BookDetailView = BookDetailView.loadFromNib()!
     private let bookDetailController = BookDetailController()
-    private var bookDetailViewModel = BookDetailViewModel(bookID: -1)
+    private var bookDetailViewModel: BookDetailViewModel
+    
+    private let loadedCommmentsSignalPipe = Signal<Bool, NoError>.pipe()
+    var loadedCommentsSignal: Signal<Bool, NoError> {
+        return loadedCommmentsSignalPipe.output
+    }
     
-    private var commentList: [Comment]
+    deinit {
+        loadedCommmentsSignalPipe.input.sendCompleted()
+    }
     
-    let dispatchGroup = DispatchGroup()
+    private var commentList: [Comment] = []
     
-    init(bookID: Int) {
-        self.bookID = bookID-1
-        self.bookDetailViewModel.bookID = bookID
-        commentList = CommentDB.getCommentsUsingBookID(bookID: bookID)
+    init(withBookDetailViewModel: BookDetailViewModel) {
+        self.bookDetailViewModel = withBookDetailViewModel
         super.init(nibName: "BookDetailViewController", bundle: Bundle.main)
     }
     
     required init?(coder aDecoder: NSCoder) {
-        self.bookID = -1
-        self.commentList = []
+        self.bookDetailViewModel = BookDetailViewModel(book: Book(status: "-1", id: -1, author: "-1", title: "-1", image: "-1", year: "-1", genre: "-1"))
```

Comment:
Remove unnecessary `self`'s. 
Also, remove that `Book` with `"-1"` values. You can either create another constructor/initializer that doesn't need a book, or perhaps modify the current one, making `book` an optional parameter.